### PR TITLE
[CARBONDATA-2815][Doc] Add documentation for spilling memory and datamap rebuild

### PR DESCRIPTION
**Pull request by xuchuanyin**
Add documentation for
1. spilling unsafe memory for data loading
2. datamap rebuild for index datamap

Be sure to do all of the following checklist to help us incorporate 
your contribution quickly and easily:

 - [ ] Any interfaces changed?
 
 - [ ] Any backward compatibility impacted?
 
 - [ ] Document update required?

 - [ ] Testing done
        Please provide details on 
        - Whether new unit test cases have been added or why no new tests are required?
        - How it is tested? Please attach test report.
        - Is it a performance related change? Please attach the performance test report.
        - Any additional information to help reviewers in testing this change.
       
 - [ ] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA.

### CODE REVIEW

File: docs/configuration-parameters.md [hunk 1]
```diff
@@ -69,7 +69,8 @@ This section provides the details of all the configurations required for CarbonD
...
+| carbon.load.sortmemory.spill.percentage | 0 | If we use unsafe memory during data loading, this configuration will be used to control the behavior of spilling inmemory pages to disk. Internally in Carbondata, during sorting carbondata will sort data in pages and add them in unsafe memory. If the memory insufficient, carbondata will spill the pages to disk and generate sort temp file. This configuration controls how many pages in memory will be spilled to disk based size. The size can be calculated by multiply this configuration value with 'carbon.sort.storage.inmemory.size.inmb'. For example, default value 0 means that no pages in unsafe memory will be spilled and all the newly sorted data will be spilled to disk; Value 50 means that if the unsafe memory is insufficient, about half of pages in the unsafe memory will be spilled to disk while value 100 means that almost all pages in unsafe memory will be spilled. **Note**: This configuration only works for 'LOCAL_SORT' and 'BATCH_SORT' and the actual spilling behavior may slightly be different in each data loading. | Integer values between 0 and 100 |
```
Comment (chetandb) [inline docs/configuration-parameters.md]: Change "memory insufficient" to "memory is insufficient".
Change "multiply" to "multiplying"

File: docs/datamap/datamap-management.md [hunk 2]
```diff
@@ -22,13 +22,13 @@ Currently, there are 5 DataMap implementation in CarbonData.
...
 | mv               | multi-table pre-aggregate table,         | No DMPROPERTY is required                | Manual           |
 | lucene           | lucene indexing for text column          | index_columns to specifying the index columns | Manual/Automatic |
-| bloom            | bloom filter for high cardinality column, geospatial column | index_columns to specifying the index columns | Manual/Automatic |
+| bloomfilter      | bloom filter for high cardinality column, geospatial column | index_columns to specifying the index columns | Manual/Automatic |
 
 ## DataMap Management
 
 There are two kinds of management semantic for DataMap.
 
-1. Autmatic Refresh: Create datamap without `WITH DEFERED REBUILD` in the statement
```
Comment (chetandb) [inline docs/datamap/datamap-management.md]: Change Autmatic to Automatic

File: docs/datamap/datamap-management.md [hunk 3]
```diff
@@ -51,15 +51,23 @@ If user do want to perform above operations on the main table, user can first dr
...
 
+We do recommend you to use this management for index datamap.
+
 ### Manual Refresh
 
 When user creates a datamap specifying maunal refresh semantic, the datamap is created with status *disabled* and query will NOT use this datamap until user can issue REBUILD DATAMAP command to build the datamap. For every REBUILD DATAMAP command, system will trigger a full rebuild of the datamap. After rebuild is done, system will change datamap status to *enabled*, so that it can be used in query rewrite.
 
-For every new data loading, data update, delete, the related datamap will be made *disabled*.
+For every new data loading, data update, delete, the related datamap will be made *disabled*,
+which means that the following queries will not benefit from the datamap before it becomes *enable* again.
```
Comment (chetandb) [inline docs/datamap/datamap-management.md]: Change enable to enabled

File: docs/datamap/datamap-management.md [hunk 4]
```diff
@@ -51,15 +51,23 @@ If user do want to perform above operations on the main table, user can first dr
...
 
 If the main table is dropped by user, the related datamap will be dropped immediately.
 
-*Note: If you are creating a datamap on external table, you need to do manual managment of the datamap.*
+**Note**:
++ If you are creating a datamap on external table, you need to do manual managment of the datamap.
```
Comment (chetandb) [inline docs/datamap/datamap-management.md]: Change managment  to management

File: docs/configuration-parameters.md (hunk 1 above)
Comment (xuchuanyin) [reply docs/configuration-parameters.md]: fixed

File: docs/datamap/datamap-management.md (hunk 2 above)
Comment (xuchuanyin) [reply docs/datamap/datamap-management.md]: fixed

File: docs/datamap/datamap-management.md (hunk 3 above)
Comment (xuchuanyin) [reply docs/datamap/datamap-management.md]: fixed

File: docs/datamap/datamap-management.md (hunk 4 above)
Comment (xuchuanyin) [reply docs/datamap/datamap-management.md]: fixed

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/7745/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/6470/

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/6133/

**Comment by QiangCai** [comment]
LGTM

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/6135/

**Pull request closed without merging**